hibernate-core/src/main/java/org/hibernate/metamodel/internal/MetadataContext.java

```diff
@@ -93,8 +93,8 @@ public MetadataContext(
 		return Collections.unmodifiableMap( entityTypes );
 	}
 
-	public Map<Class<?>, EmbeddableTypeImpl<?>> getEmbeddableTypeMap() {
-		return Collections.unmodifiableMap( embeddables );
+	public Set<EmbeddableTypeImpl<?>> getEmbeddableTypeMap() {
```

Comment:
`#getEmbeddableTypeMap` returning a `Set` is very confusing - this should change to be called something else.

Reply:
Probably a `List` would be more suitable since we don't override the default equals/hashCode in `EmbeddabeTypeImpl`

Reply:
@sebersole The `Metamodel#embeddable(Class)` should work the same since I kep the previous `Map` which holds the Class-to-EmbeddabeTyeImpl mapping. The only difference is that now we get all `EmbeddabeTyeImpl` when calling `getEmbeddables`.

I'm not sure whether this change will afect the TCK since I don't have the TCK. Is there a page which I should read in order to run the TCK?

I'll also add a bug for he JPA spec related to this issue.

Reply:
Ok, I just missed that then.  Carry on... :)